src/passes/Precompute.cpp

```diff
@@ -192,8 +192,11 @@ struct Precompute
       // TODO: handle multivalue types
       return;
     }
-    if (flow.getSingleValue().type.isVector()) {
-      return;
+    // Abort if there is a vector involved
+    for (auto type : flow.getType().expand()) {
+      if (type.isVector()) {
+        return;
+      }
```

Comment:
If this is common perhaps we should add `hasVector` alongside `isVector`? Just a thought.